cpp/src/gandiva/selection_vector.cc

```diff
@@ -26,34 +26,39 @@
 
 namespace gandiva {
 
-Status SelectionVector::PopulateFromBitMap(const uint8_t* bitmap, int bitmap_size,
-                                           int max_bitmap_index) {
+Status SelectionVector::PopulateFromBitMap(const uint8_t* bitmap, int64_t bitmap_size,
+                                           int64_t max_bitmap_index) {
   if (bitmap_size % 8 != 0) {
     std::stringstream ss;
     ss << "bitmap size " << bitmap_size << " must be padded to 64-bit size";
     return Status::Invalid(ss.str());
   }
-  if (static_cast<unsigned int>(max_bitmap_index) > GetMaxSupportedValue()) {
+  if (max_bitmap_index < 0) {
+    std::stringstream ss;
+    ss << "max bitmap index " << max_bitmap_index << " must be positive";
+    return Status::Invalid(ss.str());
+  }
```

Comment:
I found this a bit odd, should this be a DCHECK or could this arise in real code?